JULIA/p2p.jl

```diff
@@ -56,83 +56,87 @@
 #
 # *******************************************************************
 
-# ********************************************************************
-# read and test input parameters
-# ********************************************************************
-
-println("Parallel Research Kernels version ") #, PRKVERSION)
-println("Julia pipeline execution on 2D grid")
-
-if length(ARGS) != 3
-    println("argument count = ", length(ARGS))
-    println("Usage: ./synch_p2p <# iterations> <first array dimension> <second array dimension>")
-    exit(1)
-end
-
-argv = map(x->parse(Int64,x),ARGS)
+function iterate_over_grid!(grid, m, n, iterations)
+    for k = 1:iterations+1
+        for j = 2:n
+            for i = 2:m
+                @inbounds grid[i,j] = grid[i-1,j] + grid[i,j-1] - grid[i-1,j-1]
+            end
+        end
 
```

Comment:
@kpamnany Is it necessary to include the k loop here?  The more natural factorization I would expect in real application code is to have the ji loops in a function, but the k loop somewhere else.